IoTuring/Entity/Deployments/Wifi/Wifi.py

```diff
@@ -0,0 +1,433 @@
+SHOW_NA = False  # don't show not available extraAttributes
+
+KEY_SIGNAL_STRENGTH_PERCENT = "signal_strength_percent"
+KEY_SIGNAL_STRENGTH_DBM = "signal_strength_dbm"
+
+# WINDOWS
+EXTRA_KEY_NAME = "name"
+EXTRA_KEY_DESCRIPTION = "description"
+EXTRA_KEY_GUID = "guid"
+EXTRA_KEY_PHYSICAL_ADDRESS = "physical_address"
+EXTRA_KEY_STATE = "state"
+EXTRA_KEY_SSID = "ssid"
+EXTRA_KEY_BSSID = "bssid"
+EXTRA_KEY_NETWORK_TYPE = "network_type"
+EXTRA_KEY_RADIO_TYPE = "radio_type"
+EXTRA_KEY_AUTHENTICATION = "authentication"
+EXTRA_KEY_CIPHER = "cipher"
+EXTRA_KEY_CONNECTION_MODE = "connection_mode"
+EXTRA_KEY_CHANNEL = "channel"
+EXTRA_KEY_RECEIVE_RATE = "receive_rate"
+EXTRA_KEY_TRANSMIT_RATE = "transmit_rate"
+EXTRA_KEY_SIGNAL = "signal"
+EXTRA_KEY_PROFILE = "profile"
+EXTRA_KEY_HOSTED_NETWORK_STATUS = "hosted_network_status"
+# LINUX
+EXTRA_KEY_ESSID = "ESSID"
+EXTRA_KEY_MODE = "Mode"
+EXTRA_KEY_FREQUENCY = "Frequency"
+EXTRA_KEY_ACCESS_POINT = "Access_Point"
+EXTRA_KEY_BIT_RATE = "Bit_Rate"
+EXTRA_KEY_TX_POWER = "Tx_Power"
+EXTRA_KEY_LINK_QUALITY = "Link_Quality"
+EXTRA_KEY_SIGNAL_LEVEL = "Signal_Level"
+EXTRA_KEY_RX_INVALID_NWID = "Rx_invalid_nwid"
+EXTRA_KEY_RX_INVALID_CRYPT = "Rx_invalid_crypt"
+EXTRA_KEY_RX_INVALID_FRAG = "Rx_invalid_frag"
+EXTRA_KEY_TX_EXCESSIVE_RETRIES = "Tx_excessive_retries"
+EXTRA_KEY_INVALID_MISC = "Invalid_misc"
+EXTRA_KEY_MISSED_BEACON = "Missed_beacon"
+# MACOS
+EXTRA_KEY_AGRCTLRSSI = "agrCtlRSSI"
+EXTRA_KEY_AGREXTRSSI = "agrExtRSSI"
+EXTRA_KEY_AGRCTLNOISE = "agrCtlNoise"
+EXTRA_KEY_AGREXTNOISE = "agrExtNoise"
+# EXTRA_KEY_STATE = 'state' # already in windows
+EXTRA_KEY_OP_MODE = "op mode"
+EXTRA_KEY_LASTTXRATE = "lastTxRate"
+EXTRA_KEY_MAXRATE = "maxRate"
+EXTRA_KEY_LASTASSOCSTATUS = "lastAssocStatus"
+EXTRA_KEY_802_11_AUTH = "802.11 auth"
+EXTRA_KEY_LINK_AUTH = "link auth"
+# EXTRA_KEY_BSSID = 'BSSID' # already in windows
+# EXTRA_KEY_SSID = 'SSID' # already in windows
+EXTRA_KEY_MCS = "MCS"
+# EXTRA_KEY_CHANNEL = 'channel' # already in windows
+
+
+class Wifi(Entity):
+    NAME = "Wifi"
+    ALLOW_MULTI_INSTANCE = True
+
+    def Initialize(self):
+        self.platform = OsD.GetOs()
+        self.locale_str, _ = locale.getdefaultlocale()
+        self.language: str = self.locale_str.split("_")[0]
+        self.showNA = SHOW_NA
+
+        # In macos trick the language to be english since the output of airport is always in english
+        if OsD.IsMacos():
+            self.language = "en"
+
+        self.wifiInterface = self.GetFromConfigurations(CONFIG_KEY_WIFI)
+
+        self.commands = {
+            OsD.WINDOWS: ["netsh", "wlan", "show", "interfaces"],
+            OsD.LINUX: ["iwconfig", self.wifiInterface],
+            OsD.MACOS: [
+                "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport",
+                "-I",
+            ],
+        }
+        self.patterns = {
+            OsD.WINDOWS: {
+                "en": {
```

Comment:
Are you sure this is the only way to get this data? Windows 11 has support for 110 languages, are you sure you want to write this for all of them?

Reply:
My intent never was to support all languages. I'll look at other ways.